scripts/projects.js

```diff
@@ -71,39 +69,30 @@
     var parseData = JSON.parse(dataFromStorage);
     Project.loadAll(parseData);
     projectView.renderProjectPage();
-    projectView.showLanguages();
-    projectView.showLibraries();
-  };
-
-  Project.allLanguages = function() {
-    return Project.allProjects.map(function(p) {
-      return p.languages;
-    }).reduce(function(a, b) {
-      return a.concat(b);
-    },[])
-      .reduce(function(acc, cur){
-        if (acc.indexOf(cur) === -1) {
-          acc.push(cur);
-        }
-        return acc;
-      },[]);
+    projectView.show(Project.getFromProject('languages'), 'li', 'languageList');
+    projectView.show(Project.getFromProject('libraries'), 'li', 'libraryList');
+    projectView.show(Project.getFromProject('languages'), 'option', 'language-filter');
+    projectView.show(Project.getFromProject('libraries'), 'option', 'library-filter');
   };
 
-  Project.allLibraries = function() {
-    return Project.allProjects.map(function(p) {
-      return p.libraries;
-    }).reduce(function(a, b) {
-      return a.concat(b);
-    },[])
-      .reduce(function(acc, cur){
-        if (acc.indexOf(cur) === -1) {
-          acc.push(cur);
-        }
-        return acc;
-      },[]);
-
+  Project.getFromProject = function(projectKey) {
```

Comment:
Another refactored method that takes in an array from the portfolioProjects objects (Either the libraries or languages) and returns an array of unique values that are found within all projects.